openmc/polynomial.py

```diff
@@ -75,7 +75,68 @@ def order(self):
     def __call__(self, r):
         import openmc.lib as lib
         if isinstance(r, Iterable):
-            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i))
+            return [np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r_i / self.radius))
                     for r_i in r]
         else:
-            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r))
+            return np.sum(self._norm_coef * lib.calc_zn_rad(self.order, r / self.radius))
+
+
+class Zernike(Polynomial):
+    """Create Zernike polynomials given coefficients and domain.
+    The Zernike polynomials are defined as in
+    :class:`ZernikeFilter`.
+    Parameters
+    ----------
+    coef : Iterable of float
+        A list of coefficients of each term in radial only Zernike polynomials
+    radius : float
+        Domain of Zernike polynomials to be applied on. Default is 1.
+    r : float
+        Position to be evaluated, normalized on radius [0,1]
+    theta : float 
+        Azimuthal to be evaluated, normalized on :math:`[0, 2*\pi]`
```

Comment:
Remove the `r` and `theta` parameters. This section is only intended for parameters to `__init__`